### ISSUE TITLE
Update on Terrain Generation in PyCraft

### ISSUE THREAD

**Issue by HeyItsAnushtubh** [RedHenDev/python#2]
Ok, so as you said in the last issue, I did apply the new model and texture everywhere i went in the code.:sweat_smile: 
But, after the blocks combine into a megaset, the the texture is no longer there. Now, for the first time ever i found a solution. :laughing:. So wherever I saw a model without the `block` in it, I replaced it with `block` and replaced the texture to `block_texture`.
But still the issue wasnt resolved but i found the solution in this if statement. on line 183
So here it is @RedHenDev :
When we combine the megsets, 
`if subDic.get('x'+str(x)+'z'+str(z))!='i':
        subCubes[currentCube].enable()
        subCubes[currentCube].x = x
        subCubes[currentCube].z = z
        subDic['x'+str(x)+'z'+str(z)] = 'i'
        subCubes[currentCube].parent = subsets[currentSubset]
        y = subCubes[currentCube].y = genPerlin(x,z)
        g = nMap(y,-8,21,12,243)
        g += random.randint(-12,12)
        subCubes[currentCube].color = color.rgb(0,g,0)
        subCubes[currentCube].disable()
        currentCube+=1`
In the last part, I added : `
        subCubes[currentCube].model = 'block'
        subCubes[currentCube].texture = 'block_texture'  ` 
(Sorry its not indented or placed underneath eachother or something)

So Before Combine (old code)- 
![before_combine1](https://user-images.githubusercontent.com/87482637/127728999-cb665a3d-a0b2-4012-b4f8-817f3dd51dae.png)
After Combine - (old code)-
![after_combine1](https://user-images.githubusercontent.com/87482637/127729005-000253c9-9c02-491c-9e22-bcb04826d009.png)
look at the texture persistnce.....

Now Before Combine(my new code)-
![before_combine2](https://user-images.githubusercontent.com/87482637/127729023-2cfc6d71-d276-43eb-a209-a638c134371e.png)
After Combine Megasets(new code)-
![after_combine2](https://user-images.githubusercontent.com/87482637/127729040-a8561b07-bcd8-48d5-baf9-91813e926093.png)

Now look at the Texture persistence...
You can also change it in the repo, Just search for 'model' and change it to 'block' and search 'texture' to 'block_texture'

**Comment by HeyItsAnushtubh** [comment]
Ok so, you need to work on this...
Whenever The blocks combine into megasets, there's a high chance glitvhing through the terrain.
The whole whole frame comes to a pause and then suddenly fall through the terrain.

**Comment by RedHenDev** [comment]
Hi Anush

Thanks for this. [Try this](https://github.com/RedHenDev/ursina_tutorials/blob/24b1b2df921c325ec45207547ad9334743b0ae68/python_minecraft_tut_2021/pyCraft_model_ISSUE_debug.py)

I've upgraded the gravity system, and fiddled with the megaset and subset generation numbers. Textures seem to be working correctly etc. 

![Screenshot 2021-07-31 at 09 45 36](https://user-images.githubusercontent.com/24874180/127734436-c9e81659-e37c-4875-a19d-2c10fdc35fcc.png)

NB. I'm moving the python tutorials to their own dedicated repo (i.e. the link I've used above). I'm planning on changing the links in the descriptions of all videos to that repo, and won't update the old one anymore -- just so you know.

Let me know if this new code solves the issues :)

Bri
